### PR TITLE
test(doctor): doctor endpoint tests

### PR DESCRIPTION
**Pull request by jackfrancis**
Refactored `handlers.DoctorHandler` to accept a `*apiclient.WorkflowManager`, so that we can pass in mock api clients for testing.

This PR should make functional test coverage current with recent introductory deis doctor code.

### CODE REVIEW

File: handlers/handlers_test.go [hunk 1]
```diff
@@ -97,11 +99,21 @@ func TestComponentsHandler(t *testing.T) {
...
+	defer ts.Close()
+	apiClient, err := config.GetSwaggerClient(ts.URL)
 	doctorHandler := DoctorHandler(
 		mockInstalledComponents{},
 		&mockClusterID{},
 		mockAvailableVersion{},
 		data.NewFakeKubeSecretGetterCreator(nil, nil),
+		apiClient,
 	)
 	resp, err := getTestHandlerResponse(doctorHandler)
```
Comment (arschles) [inline handlers/handlers_test.go]: doesn't this start up another server for the versions.deis.com mock?

File: boot_test.go [hunk 2]
```diff
@@ -63,7 +82,12 @@ func TestGetID(t *testing.T) {
 }
 
 func testGet(route string) (*http.Response, error) {
-	server := newServer()
+	//TODO close apiServer (2nd return value from getWfmMockAPIClient)
```
Comment (arschles) [inline boot_test.go]: this still needs doin :)

File: handlers/handlers_test.go (hunk 1 above)
Comment (jackfrancis) [inline handlers/handlers_test.go]: it doesn't start one, the `apiClient` argument there *is* a `*apiclient.WorkflowManager` which has a reference to a running server

File: boot_test.go [hunk 3]
```diff
@@ -64,14 +67,20 @@ func TestGetComponents(t *testing.T) {
 
 func TestPostDoctor(t *testing.T) {
 	const doctorRoute = "/doctor"
-	resp, err := testPostNoBody(doctorRoute)
+	resp, apiServer, err := testPostNoBody(doctorRoute)
+	if apiServer != nil {
+		defer apiServer.Close()
```
Comment (arschles) [inline boot_test.go]: can you do this w/o the `defer`?

**Pull request merged**